pkg/loadbalancer/experimental/script_test.go

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package experimental
+
+import (
+	"bufio"
+	"context"
+	"maps"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/cilium/hive/cell"
+	"github.com/cilium/hive/hivetest"
+	"github.com/cilium/hive/script"
+	"github.com/cilium/hive/script/scripttest"
+	"github.com/cilium/statedb"
+	"github.com/spf13/pflag"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	daemonk8s "github.com/cilium/cilium/daemon/k8s"
+	"github.com/cilium/cilium/pkg/datapath/tables"
+	"github.com/cilium/cilium/pkg/hive"
+	"github.com/cilium/cilium/pkg/k8s/client"
+	"github.com/cilium/cilium/pkg/k8s/testutils"
+	"github.com/cilium/cilium/pkg/k8s/version"
+	"github.com/cilium/cilium/pkg/option"
+)
+
+func TestScript(t *testing.T) {
+	version.Force(testutils.DefaultVersion)
+
+	// pkg/k8s/endpoints.go uses this in ParseEndpointSlice*
+	option.Config.EnableK8sTerminatingEndpoint = true
+
+	log := hivetest.Logger(t)
+	scripttest.Test(t,
+		context.Background(),
+		func(t testing.TB) *script.Engine {
+			h := hive.New(
+				client.FakeClientCell,
+				daemonk8s.ResourcesCell,
+				Cell,
+				cell.Provide(
+					tables.NewNodeAddressTable,
+					statedb.RWTable[tables.NodeAddress].ToTable,
+					func() *option.DaemonConfig {
+						return &option.DaemonConfig{
+							EnableIPv4: true,
+							EnableIPv6: true,
+						}
+					},
+				),
+				cell.Invoke(statedb.RegisterTable[tables.NodeAddress]),
+			)
+
+			// Parse and process the "#! --flag=true" magic line.
+			_, scriptFile, _ := strings.Cut(t.Name(), "/")
+			if args := parseArgsLine("testdata/" + scriptFile); len(args) > 0 {
+				flags := pflag.NewFlagSet("", pflag.ContinueOnError)
+				h.RegisterFlags(flags)
+				require.NoError(t, flags.Parse(args), "flags.Parse")
+				t.Logf("Parsed arguments: %v", args)
+			} else {
+				// No args provided, go with the defaults.
+				hive.AddConfigOverride(h,
+					func(cfg *Config) {
+						cfg.EnableExperimentalLB = true
+					})
+			}
+			t.Cleanup(func() {
+				assert.NoError(t, h.Stop(log, context.TODO()))
+			})
+			cmds, err := h.ScriptCommands(log)
+			require.NoError(t, err, "ScriptCommands")
+			maps.Insert(cmds, maps.All(script.DefaultCmds()))
+
+			return &script.Engine{
+				Cmds: cmds,
+			}
+		}, []string{}, "testdata/*.txtar")
+}
+
+func parseArgsLine(file string) []string {
```

Comment:
Wonder if this should be made into a builtin feature of scripttest and the args passed to the engine constructor. Not in this PR though.

Reply:
Done in https://github.com/cilium/hive/pull/22